Create cn

Boldan opened the pull request.

Mercury13 commented:
Oh, what do you want? Auto-translate? IDK, English and Chinese are so far apart that auto-translation will make so many funny moments. Manually translate? Go on, I’ll happily help in every technical aspect and even write more of Unicodia.

Boldan commented:
> Oh, what do you want? Auto-translate? IDK, English and Chinese are so far apart that auto-translation will make so many funny moments. Manually translate? Go on, I’ll happily help in every technical aspect and even write more of Unicodia.

The Chinese translation file has been uploaded to the Lang-src directory. Since I don't know what to do next, can you help me with it? Thank you very much.

Mercury13 commented:
I see 1-byte long file named “cn”. What have you translated with? Where is translation?

Mercury13 commented:
Maybe you can attach your translations right here, and I’ll think what to do?

Boldan commented:
> Maybe you can attach your translations right here, and I’ll think what to do?

![Screenshot_20250214_025447.jpg](https://github.com/user-attachments/assets/f60fc65b-8cbe-4824-a3bd-e35e5aec2649)

https://github.com/Boldan/unicodia/tree/main/Lang-src

Boldan commented:
> Maybe you can attach your translations right here, and I’ll think what to do?

I have fork synchronization,I don't know if you can see it in the same directory.

Mercury13 commented:
So, you’ll use UTranslator. That’s a program that supports translating a living program whose texts change frequently.

Mercury13 commented:
For now I don’t have time to write a converter from INI to UTranslator.
Moreover, I’m not developing at all (and will not until Sunday).
Which version have you taken lang.ini from? — I’ll build a *.utran from that *.uorig and your ini…
If the version is recent enough, just check once again these strings.
Blocks.1F800.Text
Blocks.1FA00.Text
Lib.Group.OnOff
Term.egyp.Text
Version.17.0.Text
Version.Bullet.[all]
GlyphVar.Vert.[all]
OldComp.BbcMaster.Text

Mercury13 commented:
I’ve taken your locale.xml, but we’ll discuss alphabetical sorting separately.
How should A-Z icon look like?
![image](https://github.com/user-attachments/assets/b7d67a75-3a08-4347-b2d3-8b8607d327c8)
And which are your Chinese principles of alphabetical sorting? Maybe we’ll invent something, like long ago we invented together with a Japanese.

Mercury13 commented:
Again, I’ll write a converter later, there are a few obstacles but they are probably possible to overcome.
**Which version have you taken lang.ini from?** — it’s important for me.

Boldan commented:
> For now I don’t have time to write a converter from INI to UTranslator.
> Moreover, I’m not developing at all (and will not until Sunday).
> Which version have you taken lang.ini from? — I’ll build a *.utran from that *.uorig and your ini…
> If the version is recent enough, just check once again these strings.
> Blocks.1F800.Text
> Blocks.1FA00.Text
> Lib.Group.OnOff
> Term.egyp.Text
> Version.17.0.Text
> Version.Bullet.[all]
> GlyphVar.Vert.[all]
> OldComp.BbcMaster.Text

version 16

Mercury13 commented:
Version of Unicodia. Something like 2.##.##.

Mercury13 commented:
> version 16

I mean version of Unicodia. Probably around 2.9.#, but what is the last digit?

Boldan commented:
> I’ve taken your locale.xml, but we’ll discuss alphabetical sorting separately.
> How should A-Z icon look like?
> ![image](https://github.com/user-attachments/assets/b7d67a75-3a08-4347-b2d3-8b8607d327c8)
> And which are your Chinese principles of alphabetical sorting? Maybe we’ll invent something, like long ago we invented together with a Japanese.

There are two common types of Chinese sorting:
1. How many strokes of Chinese characters
2. Convert Chinese characters to pinyin first, and then sort them by letter

Mercury13 commented:
> There are two common types of Chinese sorting:
>     1. How many strokes of Chinese characters
>     2. Convert Chinese characters to pinyin first, and then sort them by letter

Anyway it’s difficult, and I can sort first by # of strokes, then by radical. Is it okay? It’s Unicodia, we’ll just look into bases, but for now I don’t have total # of strokes.

Mercury13 commented:
Oh, your version is rather new, so I’ll find the 1st version that has both 17.0 and BBC Master, and you’ll check what changed. Afterwards, I won’t develop until Sunday.

Boldan commented:
> Again, I’ll write a converter later, there are a few obstacles but they are probably possible to overcome.
> **Which version have you taken lang.ini from?** — it’s important for me.

The initial version installed was 2.9.1, and then updated to 2.10.1 yesterday. The translation file is based on the English version of the 2.10.1 version of lan.ini

Mercury13 commented:
It’s interesting, and anyway I need time to write a converter.
Did you retranslate if original’s string changed? — I changed lots of strings, both because of English grammar and Unicode 17.0.
Which language did you translate from? Japanese?

Boldan commented:
There is also a relatively simple sorting method:The unicode of Chinese characters is sorted from small to large, because Chinese characters are generally added to unicode according to the frequency of use of Chinese characters



---Original---
From: ***@***.***&gt;
Date: Fri, Feb 14, 2025 03:48 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Mercury13/unicodia] Create cn (PR #457)


    
There are two common types of Chinese sorting:
 1. How many strokes of Chinese characters
 2. Convert Chinese characters to pinyin first, and then sort them by letter
  
Anyway it’s difficult, and I can sort first by # of strokes, then by radical. Is it okay? It’s Unicodia, we’ll just look into bases, but for now I don’t have total # of strokes.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  Mercury13 left a comment (Mercury13/unicodia#457)
  
There are two common types of Chinese sorting:
 1. How many strokes of Chinese characters
 2. Convert Chinese characters to pinyin first, and then sort them by letter
  
Anyway it’s difficult, and I can sort first by # of strokes, then by radical. Is it okay? It’s Unicodia, we’ll just look into bases, but for now I don’t have total # of strokes.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

Boldan commented:
No, no, no, Chinese



---Original---
From: ***@***.***&gt;
Date: Fri, Feb 14, 2025 03:57 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Mercury13/unicodia] Create cn (PR #457)


   
It’s interesting, and anyway I need time to write a converter.
 Did you retranslate if original’s string changed? — I changed lots of strings, both because of English grammar and Unicode 17.0.
 Which language did you translate from? Japanese?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  Mercury13 left a comment (Mercury13/unicodia#457)
 
It’s interesting, and anyway I need time to write a converter.
 Did you retranslate if original’s string changed? — I changed lots of strings, both because of English grammar and Unicode 17.0.
 Which language did you translate from? Japanese?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

Boldan commented:
> It’s interesting, and anyway I need time to write a converter.
> Did you retranslate if original’s string changed? — I changed lots of strings, both because of English grammar and Unicode 17.0.
> Which language did you translate from? Japanese?

No, no, no, the .ini file is translated from English，The .xml file is borrowed from the Japanese because there's a lot in common

Boldan commented:
> > There are two common types of Chinese sorting:
> >     1. How many strokes of Chinese characters
> >     2. Convert Chinese characters to pinyin first, and then sort them by letter
> 
> Anyway it’s difficult, and I can sort first by # of strokes, then by radical. Is it okay? It’s Unicodia, we’ll just look into bases, but for now I don’t have total # of strokes.

I'm almost dawn here. I'm too tired to sleep for a while. I'll give you a file of Chinese character strokes when I get up

Mercury13 commented:
Did you retranslate the strings if they changed? Changelist 2.9.1→2.10.1 is rather long.

Mercury13 commented:
Oh, I invented a way. Assemble a lockit of 2.9.1, update with 2.10.1, and then check with auto-translator whether the string was touched. Afterwards, probably on Sunday, have a more urgent job (“develop” photos for a theatre bill).

Boldan commented:
> Oh, I invented a way. Assemble a lockit of 2.9.1, update with 2.10.1, and then check with auto-translator whether the string was touched. Afterwards, probably on Sunday, have a more urgent job (“develop” photos for a theatre bill).

OK, this is not in a hurry. You'll deal with the important things first

Boldan commented:
I have uploaded the Chinese translation file in this directory (https://github.com/Boldan/unicodia/tree/main/Lang-src), you can copy it

Boldan commented:
> I have uploaded the Chinese translation file in this directory (https://github.com/Boldan/unicodia/tree/main/Lang-src), you can copy it

![Screenshot_20250214_195037.jpg](https://github.com/user-attachments/assets/b9b2a6d3-fc5f-4c74-a819-b4b102055eb8)

Mercury13 commented:
As far as I understand, you’ve translated 2.9.1 using AI, and when 2.10.1 came, you did the same. Right?

Boldan commented:
Yeah, but I'll check it myself



---Original---
From: ***@***.***&gt;
Date: Sun, Feb 16, 2025 07:49 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [Mercury13/unicodia] Create cn (PR #457)


   
As far as I understand, you’ve translated 2.9.1 using AI, and when 2.10.1 came, you did the same. Right?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  Mercury13 left a comment (Mercury13/unicodia#457)
 
As far as I understand, you’ve translated 2.9.1 using AI, and when 2.10.1 came, you did the same. Right?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;

Mercury13 commented:
I hand-accepted your changes. What’s needed next.
1. There is some wiki markup. At least version info (17.0 etc) destroyed hyperlinks.
2. I marked two texts as badly translated — they are INCOMPLETE (forgot one paragraph).

Mercury13 commented:
3. What to write in Credits?

Mercury13 commented:
4. {{_}} = non-break space. In Chinese there’s probably another place for non-break space.
(You just save the translation with UTranslator and automatically get lang.ini!)

Mercury13 commented:
5. What to draw on Chinese version of A-Z icon?
![image](https://github.com/user-attachments/assets/67df8f4a-b627-4517-af57-375bbfaa145d)
![image](https://github.com/user-attachments/assets/cbf28ccb-239b-4466-b663-4ce715480fd2)
![image](https://github.com/user-attachments/assets/d2d01809-ca4b-445a-aaaf-c8365e51c985)
Drew for these languages — will draw for Chinese too…

Mercury13 commented:
5… Decided to sort by # of strokes, as it’s at least present in Unicode bases

Mercury13 commented:
Even if I reject the PR, believe me, I already hand-accepted.

Mercury13 commented:
You are really inventive, and I quickly integrated your work.

Boldan commented:
> 5. What to draw on Chinese version of A-Z icon?
> ![image](https://github.com/user-attachments/assets/67df8f4a-b627-4517-af57-375bbfaa145d)
> ![image](https://github.com/user-attachments/assets/cbf28ccb-239b-4466-b663-4ce715480fd2)
> ![image](https://github.com/user-attachments/assets/d2d01809-ca4b-445a-aaaf-c8365e51c985)
> Drew for these languages — will draw for Chinese too…

You can use "笔画" as the icon

Boldan commented:
> 4. {{_}} = non-break space. In Chinese there’s probably another place for non-break space.
> (You just save the translation with UTranslator and automatically get lang.ini!)

Okay, I'll check it again more carefully

Boldan commented:
> I hand-accepted your changes. What’s needed next.
> 1. There is some wiki markup. At least version info (17.0 etc) destroyed hyperlinks.
> 2. I marked two texts as badly translated — they are INCOMPLETE (forgot one paragraph).

Later today, I'll fix it

Mercury13 commented:
Just 笔画 as an icon? Not something like 一凴? (IDK what ideograph should be the last)

Boldan commented:
> Just 笔画 as an icon? Not something like 一凴? (IDK what ideograph should be the last)

1. Apart from programmers, most Chinese people do not understand the meaning of "一凴"
2. "笔画" is the simplest, most direct, and most effective expression

Boldan commented:
> (You see that all three icons consist of the 1st character and the last one)

In China, it's not like that. Because there is no first word or last word in Chinese characters. "一凴" is based on English

Mercury13 commented:
Well, maybe some character of 3 strokes, and draw those strokes with red, violet and blue?

Mercury13 commented:
I cannot fit 笔画 into 16×16 icon :(

Mercury13 commented:
If something like 一九?

Boldan commented:
> What if something like 一九?

Hold on. I'll try it

Mercury13 commented:
That’s an image on icon. In menu we may write everything we want.

Boldan commented:
> That’s an image on icon. In menu we may write everything we want.

OK, I've made some icons, but they don't look beautiful😅. Just do what you want

Boldan commented:
What does "{{%|100}}" mean?
{{%|100}}＝100%？

Mercury13 commented:
Have you ever edited Wikipedia? It’s template meaning 100 + some narrow space + percent.

Mercury13 commented:
Read BuiltInWikiTemplates.txt

Boldan commented:
> Have you ever edited Wikipedia? It’s template meaning 100 + some narrow space + %.

No, I haven't used Wikipedia again because I can't access it normally

Mercury13 commented:
Here’s my design
![image](https://github.com/user-attachments/assets/8d0c78fd-147d-45d4-af4e-85875185b92a)

Boldan commented:
> Here’s my design
> ![image](https://github.com/user-attachments/assets/8d0c78fd-147d-45d4-af4e-85875185b92a)
> 

Good. It looks good

Mercury13 commented:
Well, let’s settle on 1…9.

Mercury13 commented:
If you’ve done something — open another PR, or give me a sigal.

Mercury13 commented:
@Boldan Progress?

Boldan commented:
> @Boldan Progress?

I'm sorry, he can't continue because he's got serious physical and mental problems, and he may even die. No need to reply.

Mercury13 commented:
Then closing. Sorry.
